copycat/runtime/src/main/java/org/apache/kafka/copycat/runtime/rest/RestServer.java

```diff
@@ -0,0 +1,228 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.copycat.runtime.rest;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider;
+import org.apache.kafka.copycat.errors.CopycatException;
+import org.apache.kafka.copycat.runtime.Herder;
+import org.apache.kafka.copycat.runtime.WorkerConfig;
+import org.apache.kafka.copycat.runtime.rest.entities.ErrorMessage;
+import org.apache.kafka.copycat.runtime.rest.errors.CopycatExceptionMapper;
+import org.apache.kafka.copycat.runtime.rest.errors.CopycatRestException;
+import org.apache.kafka.copycat.runtime.rest.resources.ConnectorsResource;
+import org.apache.kafka.copycat.runtime.rest.resources.RootResource;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.NetworkTrafficServerConnector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.Slf4jRequestLog;
+import org.eclipse.jetty.server.handler.DefaultHandler;
+import org.eclipse.jetty.server.handler.HandlerCollection;
+import org.eclipse.jetty.server.handler.RequestLogHandler;
+import org.eclipse.jetty.server.handler.StatisticsHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Embedded server for the REST API that provides the control plane for Copycat workers.
+ */
+public class RestServer {
+    private static final Logger log = LoggerFactory.getLogger(RestServer.class);
+
+    private static final long GRACEFUL_SHUTDOWN_TIMEOUT_MS = 60 * 1000;
+
+    private static final ObjectMapper JSON_SERDE = new ObjectMapper();
+
+    private final WorkerConfig config;
+    private Herder herder;
+    private Server jettyServer;
+
+    /**
+     * Create a REST server for this herder using the specified configs.
+     */
+    public RestServer(WorkerConfig config) {
+        this.config = config;
+
+        // To make the advertised port available immediately, we need to do some configuration here
+        String hostname = config.getString(WorkerConfig.REST_HOST_NAME_CONFIG);
+        Integer port = config.getInt(WorkerConfig.REST_PORT_CONFIG);
+
+        jettyServer = new Server();
+
+        NetworkTrafficServerConnector connector = new NetworkTrafficServerConnector(jettyServer);
```

Comment:
Yeah, probably less relevant here. We use this in Confluent's rest-utils because you can grab stats like the number of connections made and report them through metrics. Swapped out for plain old ServerConnector.